### DISCUSSION TITLE
Any chance in increase recognition on this crappy watermeter?

### DISCUSSION THREAD

**Discussion by steviehs** [jomjol/AI-on-the-edge-device#1194] [Q&A]
![grafik](https://user-images.githubusercontent.com/26877141/196932420-f5296b25-75f3-41bc-8c2e-0344c2d7d9a0.png)

This sensus 620 is really crap. Even my Human Intelligence struggles to read this...  So in every around 10th recognition is right, but e.g. right now I guess it wont work for the next 100l because the 7 is interpreted badly...

Is there anything I could do to improve or just accept it?

```
[MakeImage]
;LogImageLocation = /log/source
WaitBeforeTakingPicture = 5
;LogfileRetentionInDays = 15
Brightness = -2
Contrast = 2
Saturation = 0
LEDIntensity = 65
ImageQuality = 12
ImageSize = VGA
FixedExposure = false

[Alignment]
InitialRotate = 42
InitialMirror = false
SearchFieldX = 20
SearchFieldY = 20
;AlignmentAlgo = default
FlipImageSize = false
/config/ref0.jpg 170 65
/config/ref1.jpg 438 167

[Digits]
Model = /config/dig-class100-0130_s2_q.tflite
CNNGoodThreshold = 0.5
LogImageLocation = /log/digit
LogfileRetentionInDays = 3
main.dig1 164 153 28 47 0
main.dig2 202 153 28 47 0
main.dig3 238 153 28 47 0
main.dig4 274 153 28 47 0
main.dig5 312 153 28 47 0
main.dig6 345 153 28 47 0
main.dig7 375 153 28 47 0
main.dig8 405 153 28 47 0

;[Analog]
Model = /config/ana-cont_11.3.0_s2.tflite
;LogImageLocation = /log/analog
;LogfileRetentionInDays = 3
null.ana1 432 230 92 92 false
null.ana2 379 332 92 92 false
null.ana3 283 374 92 92 false
null.ana4 155 328 92 92 false

[PostProcessing]
main.DecimalShift = -3
null.DecimalShift = 0
PreValueUse = true
PreValueAgeStartup = 720
AllowNegativeRates = false
main.MaxRateValue = 0.1
null.MaxRateValue = 0.05
main.MaxRateType = RateChange
;null.MaxRateType = AbsoluteChange
;main.ExtendedResolution = true
;null.ExtendedResolution = false
;main.IgnoreLeadingNaN = true
;null.IgnoreLeadingNaN = true
ErrorMessage = true
CheckDigitIncreaseConsistency = true

[MQTT]
Uri = mqtt://192.168.243.2:1883
MainTopic = wasserzaehler
;ClientID = wasser
;user = USERNAME
;password = PASSWORD
;SetRetainFlag = true

;[InfluxDB]
;Uri = undefined
;Database = 
;Measurement = undefined
;user = undefined
;password = undefined

;[GPIO]
;IO0 = input disabled 10 false false 
;IO1 = input disabled 10 false false 
;IO3 = input disabled 10 false false 
;IO4 = built-in-led disabled 10 false false 
;IO12 = input-pullup disabled 10 false false 
;IO13 = input-pullup disabled 10 false false 
LEDType = WS2812
LEDNumbers = 2
LEDColor = 150 150 150

[AutoTimer]
AutoStart = true
Intervall = 4.85

[Debug]
Logfile = false
LogfileRetentionInDays = 3

[System]
TimeZone = CET-1CEST,M3.5.0,M10.5.0/3
;TimeServer = undefined
;AutoAdjustSummertime = false
;Hostname = undefined
SetupMode = false
```

**Comment by ghost** [comment]
Could you add a screenshot of the recognition?

I would try to reduce the height of the ROIs at bottom (uncheck lock aspect ratio).

Else you could give us image data. It's easy with  https://github.com/haverland/collectmeterdigits. The Readme describes the steps.

**Reply by steviehs** [reply]
Sorry for the delay. Here is a screenshot of the recognition, I will try to reduce the height of the ROIs now.
![grafik](https://user-images.githubusercontent.com/26877141/202298366-97381357-a5f5-441e-91ff-d739a3af2a30.png)

**Reply by steviehs** [reply]
decreasing the height did not bring better results, I guess I will have to get rid of the reflections at digit 4 and 5... or at the end accept that this water meter is real crap :-)
![grafik](https://user-images.githubusercontent.com/26877141/202380223-e1f10e30-07e0-4130-bce2-e1665fff40fb.png)

**Reply by caco3** [reply]
did you try it with a diffuser?

See https://github.com/jomjol/AI-on-the-edge-device/discussions/984#discussioncomment-3579081

That worked for me

**Comment by Bens0n68** [comment]
It looks like you are using the built-in LED flash. There are reflections at the lower part of the digits, exactly where the differentiating information might be.
I could imagine that using external LEDs and removing the reflections could improve the rate (not fixing it, but improving).
